client/extensions/woocommerce/app/products/product-form-variations-card.js

```diff
@@ -22,7 +22,21 @@ const ProductFormVariationsCard = ( {
 } ) => {
 	const handleToggle = () => {
 		if ( 'variable' !== product.type ) {
-			editProduct( product, { type: 'variable', dimensions: {} } );
+			// Reset any 'simple' product settings that have been filled out,
```

Comment:
Actually I guess I could store that data in local state and reapply it to the Redux state when they toggle. 

Maybe it's a good idea to remove these lines for now and do that in a separte PR.